tools/swift-format/Sources/Rules/NoCasesWithOnlyFallthrough.swift

```diff
@@ -11,5 +11,97 @@ import SwiftSyntax
+    var violations: [SwitchCaseLabelSyntax] = []
+    
+    for switchCase in node.cases {
+      guard let switchCase = switchCase as? SwitchCaseSyntax else { continue }
+      guard let label = switchCase.label as? SwitchCaseLabelSyntax else { newCases.append(switchCase); continue }
+      
+      if switchCase.statements.count == 1,
+        let only = switchCase.statements.first,
+        only.item is FallthroughStmtSyntax {
+        violations.append(label)
+      } else {
+        guard violations.count > 0 else { newCases.append(switchCase); continue }
+        if retrieveNumericCaseValue(caseLabel: label) > 0 {
+          let newCase = collapseIntegerCases(violations: violations, validCaseLabel: label, validCase: switchCase)
+          newCases.append(newCase)
+        } else {
+          let newCase = collapseNonIntegerCases(violations: violations, validCaseLabel: label, validCase: switchCase)
+          newCases.append(newCase)
+        }
+        violations = []
+      }
+    }
+    return node.withCases(SyntaxFactory.makeSwitchCaseList(newCases))
+  }
+  
+  // Puts all given cases on one line with range operator or commas
+  func collapseIntegerCases(violations: [SwitchCaseLabelSyntax], validCaseLabel: SwitchCaseLabelSyntax, validCase: SwitchCaseSyntax) -> SwitchCaseSyntax {
+    var isConsecutive = true
+    var index = 0
+    var caseNums: [Int] = []
+    
+    for item in violations { caseNums.append(retrieveNumericCaseValue(caseLabel: item)) }
+    caseNums.append(retrieveNumericCaseValue(caseLabel: validCaseLabel))
+    
+    while index <= caseNums.count - 2, isConsecutive {
+      isConsecutive = caseNums[index] + 1 == caseNums[index + 1]
+      index += 1
+    }
 
+    var newCaseItems: [CaseItemSyntax] = []
+    let first = caseNums[0]
+    let last = caseNums[caseNums.count - 1]
+    if isConsecutive {
+      // Create a case with a sequence expression based on the new range
+      let start = SyntaxFactory.makeIntegerLiteralExpr(digits: SyntaxFactory.makeIntegerLiteral("\(first)"))
+      let end = SyntaxFactory.makeIntegerLiteralExpr(digits: SyntaxFactory.makeIntegerLiteral("\(last)"))
+      let newExpList = SyntaxFactory.makeExprList([start, SyntaxFactory.makeBinaryOperatorExpr(operatorToken: SyntaxFactory.makeUnspacedBinaryOperator("...")), end])
+      let newExpPat = SyntaxFactory.makeExpressionPattern(expression: SyntaxFactory.makeSequenceExpr(elements: newExpList))
+      newCaseItems.append(SyntaxFactory.makeCaseItem(pattern: newExpPat, whereClause: nil, trailingComma: nil))
+    } else {
+      // Add each case item separated by a comma
+      for num in caseNums {
+        let newExpPat = SyntaxFactory.makeExpressionPattern(expression: SyntaxFactory.makeIntegerLiteralExpr(digits: SyntaxFactory.makeIntegerLiteral("\(num)")))
+        if num != last {
+          let newCaseItem = SyntaxFactory.makeCaseItem(pattern: newExpPat, whereClause: nil, trailingComma: SyntaxFactory.makeCommaToken().withOneTrailingSpace())
```

Comment:
This could be simpler if you created one `trailingComma` token and chose it `if num != last`.

```swift
let trailingComma = SyntaxFactory.makeCommaToken(trailingTrivia: .spaces(1)
for num in caseNums {
  // existing code...
  let newCaseItem = SyntaxFactory.makeCaseItem(
    pattern: newExpPat,
    whereClause: nil,
    trailingComma: num == last ? nil : trailingComma
  )
  newCaseItems.append(newCaseItem)
}